### PR TITLE
Changed Public endpoint publicEndpoint from an attribute to init parameter

### PR DESCRIPTION
**Pull request by alexamies**
To enable application developers to set the value of publicEndpoint more easily, changed from a servlet context attribute to an init parameter.

### CODE REVIEW

File: contrib/http_servlet/src/test/java/io/opencensus/contrib/http/servlet/OcHttpServletFilterTest.java [hunk 1]
```diff
@@ -93,17 +92,16 @@ public void testInit() throws IOException, ServletException {
     when(mockConfig.getServletContext()).thenReturn(mockServletContext);
     when(mockServletContext.getAttribute(OC_TRACE_PROPAGATOR)).thenReturn(null);
     when(mockServletContext.getAttribute(OC_EXTRACTOR)).thenReturn(null);
-    when(mockServletContext.getAttribute(OC_PUBLIC_ENDPOINT)).thenReturn(null);
+    when(mockServletContext.getInitParameter(OC_PUBLIC_ENDPOINT)).thenReturn("false");
```
Comment (rghetia) [inline contrib/http_servlet/src/test/java/io/opencensus/contrib/http/servlet/OcHttpServletFilterTest.java]: Change "false" to null here and add following line after line 100
verify(mockServletContext).getInitParameter()

This is to verify that getInitParameter() is called once.

File: contrib/http_servlet/src/test/java/io/opencensus/contrib/http/servlet/OcHttpServletFilterTest.java [hunk 2]
```diff
@@ -116,17 +114,16 @@ public void testInitWithOptions() throws IOException, ServletException {
...
     when(mockServletContext.getAttribute(OC_TRACE_PROPAGATOR)).thenReturn(b3Propagator);
     when(mockServletContext.getAttribute(OC_EXTRACTOR)).thenReturn(customExtractor);
-    when(mockServletContext.getAttribute(OC_PUBLIC_ENDPOINT)).thenReturn(publicEndpoint);
+    when(mockServletContext.getInitParameter(OC_PUBLIC_ENDPOINT)).thenReturn("false");
 
     filter.init(mockConfig);
 
     verify(mockConfig).getServletContext();
-    verify(mockServletContext, times(3)).getAttribute(stringArgumentCaptor.capture());
+    verify(mockServletContext, times(2)).getAttribute(stringArgumentCaptor.capture());
```
Comment (rghetia) [inline contrib/http_servlet/src/test/java/io/opencensus/contrib/http/servlet/OcHttpServletFilterTest.java]: Add verfity(mockServletContext).getInitParameters() here as well.

File: contrib/http_servlet/src/test/java/io/opencensus/contrib/http/servlet/OcHttpServletFilterTest.java (hunk 1 above)
Comment (alexamies) [reply contrib/http_servlet/src/test/java/io/opencensus/contrib/http/servlet/OcHttpServletFilterTest.java]: Changed as suggested in latest push

File: contrib/http_servlet/src/test/java/io/opencensus/contrib/http/servlet/OcHttpServletFilterTest.java (hunk 2 above)
Comment (alexamies) [reply contrib/http_servlet/src/test/java/io/opencensus/contrib/http/servlet/OcHttpServletFilterTest.java]: Changed as suggested in latest push

**Pull request merged**